### PR TITLE
Add RFCS

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
The goal is to extract some issue that are used as a technical specification to have them persistent on the project

### CODE REVIEW

**Comment by FirelightFlagboy** [comment]
Current the RFCs are extract from some issue.
If you see that an important issue wasn't convert => write a comment ;).
If you see that a converted RFC should not be added => open a discussion :shrug:.

File: docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Proper Rust <-> Javascript number conversion in the bindings
+
+From [ISSUE-3941](https://github.com/Scille/parsec-cloud/issues/3941)
+
+Currently we only support `i32` type as lower common denominator for number.
```
Comment (TimeEngineer) [inline docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md:5]: This is wrong, we also support `u32`

**Review by TimeEngineer — CHANGES_REQUESTED** [review]

File: docs/rfcs/0000-how-to-write-rfcs.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# How to write RFCs
+
+This document indicate the `do` & `don't` when writing a RFC file.
```
Comment (mmmarcos) [inline docs/rfcs/0000-how-to-write-rfcs.md:3]:
```suggestion
This document indicate the `do` & `don't` when writing an RFC file.
```

File: docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md (hunk 1 above)
Comment (FirelightFlagboy) [reply docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md:5]: Does this please you ?

```suggestion
Currently we only support `i32` & `u32` types as lower common denominator for number.
```

File: docs/rfcs/0000-how-to-write-rfcs.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+- For pull request
+
+  ```markdown
+  [PR-NNNN](https://github.com/foo/bar/pull/NNNN)
+  ```
+
+## Suggested vscode extensions
+
+I recommend using those vscode extensions:
```
Comment (mmmarcos) [inline docs/rfcs/0000-how-to-write-rfcs.md:54]:
```suggestion
I recommend using these vscode extensions:
```

File: docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md (hunk 1 above)
Comment (TimeEngineer) [reply docs/rfcs/0036-rust-to-js-number-conversion-in-the-bindings.md:5]: Yes

**Review by mmmarcos — COMMENTED** [review]
C'est pas mal ! même si à ce stade je ne suis pas capable de reviewer le contenu des RFC existants ^^

**Review by TimeEngineer — APPROVED** [review]

File: docs/rfcs/0037-shamir-based-recovery.md [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+
+## 1 - Create&update the Shamir recovery
+
+Authenticated API:
+
+```json5
+[
+    {
+        "major_versions": [
+            4  // TODO: we most likely want to do an API bump for this feature !
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0037-shamir-based-recovery.md:61]: Those are rookie number, we need to pump them up !

File: docs/rfcs/0037-shamir-based-recovery.md [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
+                        "name": "reveal_token",
+                        "type": "Bytes"
+                    },
+                    {
+                        // Configuration is provide as a `ShamirRecoveryBriefCertificate`, it
+                        // contains the threshold for the quorum and for who each share is for
+                        // This field has a certain level of duplication with the shares one,
+                        // this is because they are used for different thing (we provide the
+                        // encrypted share data only when needed)
+                        "name": "brief",
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0037-shamir-based-recovery.md:115]: Yes, the server will store this certificate

File: docs/rfcs/0037-shamir-based-recovery.md [hunk 6]
```diff
@@ -0,0 +1,691 @@
...
+                ]
+            }
+        ]
+    }
+]
+```
+
+Consistency between `brief` and `shares` must be checked by the parsec server:
+
+- the number of shares must be greater or equal to the threshold (and threshold must be >= 1)
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0037-shamir-based-recovery.md:133]: I don't understand this change: we are listing the thing that are enforced by the server, user only allowed to have a single share would be something to write, but user able to have multiple shares is the default (and it's already mentioned earlier)

File: docs/rfcs/0037-shamir-based-recovery.md [hunk 7]
```diff
@@ -0,0 +1,691 @@
...
+        "reps": [
+            {
+                "status": "ok",
+                "other_fields": [
+                    {
+                        "name": "others",
+                        // Contains a list of `ShamirRecoveryBriefCertificate`
+                        // Depending of organization configuration, this can contains all
+                        // the Shamir recovery the current user is a recipient, or
+                        "type": "Vec<Bytes>"
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply docs/rfcs/0037-shamir-based-recovery.md:271]: This was related to **future evolution 3**, I removed the comment then

File: docs/rfcs/0037-shamir-based-recovery.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply docs/rfcs/0037-shamir-based-recovery.md:115]: :+1:

File: docs/rfcs/0037-shamir-based-recovery.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply docs/rfcs/0037-shamir-based-recovery.md:133]: It's just a suggestion, I find it more clear with it

**Review by mmmarcos — APPROVED** [review]
:+1:

**Pull request merged**